### PR TITLE
Inline and add noexcept

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

File: storage/src/vespa/storage/bucketdb/storbucketdb.h [hunk 1]
```diff
@@ -57,24 +57,20 @@ class StorBucketDatabase {
...
+    void for_each(std::function<Decision(uint64_t, const Entry &)> func, const char* clientId);
 
     [[nodiscard]] std::unique_ptr<bucketdb::ReadGuard<Entry>> acquire_read_guard() const;
 
     /**
      * Returns true iff bucket has no superbuckets or sub-buckets in the
      * database. Usage assumption is that any operation that can cause the
      * bucket to become inconsistent will require taking its lock, so by
-     * requiring the lock to be provided here we avoid race conditions.
+     * requiring the lock to be provided here we avoid race conditions.q
```
Comment (vekterli) [inline storage/src/vespa/storage/bucketdb/storbucketdb.h:73]: Spurious extra character

**Review by vekterli — DISMISSED** [review]
:+1:

File: storage/src/vespa/storage/bucketdb/storbucketdb.h (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/bucketdb/storbucketdb.h]: Fixed

File: storage/src/vespa/storage/bucketdb/storagebucketinfo.h [hunk 2]
```diff
@@ -10,23 +10,21 @@ struct StorageBucketInfo {
...
-        info.setMetaCount(1);
-        info.setDocumentCount(0);
-        info.setTotalDocumentSize(0);
+    bool valid() const noexcept { return info.valid(); }
+    void setBucketInfo(const api::BucketInfo& i) noexcept { info = i; }
+    const api::BucketInfo& getBucketInfo() const noexcept { return info; }
+    bool verifyLegal() const noexcept { return true; }
+    uint32_t getMetaCount() const noexcept { return info.getMetaCount(); }
+    bool operator == (const StorageBucketInfo &) const noexcept {
+        return true;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/bucketdb/storagebucketinfo.h]: Usage was only from similar methods in BTreeLockableMap, which turned out to also be unused.
All of them removed.

**Pull request merged**